File: evd_ros_backend/evd_ros_core/src/evd_script/program/tasks/close_gripper.py
from ..task import Task
from ..primitives import Gripper


class CloseGripper(Task):

    '''
    Data structure methods
    '''

    def __init__(self, position=100, effort=100, speed=100, thing_uuid=None, type='',
                 name='', uuid=None, parent=None, append_type=True, primitives=None):

        if primitives == None:
            primitives=[
                Gripper(
                    thing_uuid=thing_uuid,
                    position=position,
                    effort=effort,
                    speed=speed)
            ]

        super(CloseGripper,self).__init__(
            type='close-gripper.'+type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type,
            primitives=primitives)

File: evd_authoring_app/src/model/evdScript/program/tasks/closeGripper.js
import { Task } from '../task';
import { Gripper } from '../primitives';
import { NodeParser } from '../../utilityFunctions';


export class CloseGripper extends Task {

    /*
    * Data structure methods
    */

    static typeString() {
        return 'close-gripper';
    }

    static fullTypeString() {
        return Task.fullTypeString() + CloseGripper.typeString();
    }

    constructor(thingUuid=null, position=100, effort=100, speed=100, 
                type='', name='', uuid=null, parent=null, appendType=true, 
                primitives=null) 
    {

        if (primitives !== null) {
            primitives = [
                Gripper(
                    thingUuid,
                    position,
                    effort,
                    speed
                )
            ];
        }

        super(
            primitives,
            (appendType) ? 'close-gripper.'+type : type,
            name,
            uuid,
            parent,
            appendType
        );
    }

    static fromDict(dct) {
        return new CloseGripper(
            null,
            100,
            100,
            100,
            dct.type,
            dct.name,
            dct.uuid,
            false,
            dct.primitives.map(p => NodeParser(p))
        );
    }
}